### PR TITLE
Renamed dbms.allow_format_migration to dbms.allow_upgrade for clarification

### PR DESCRIPTION
**Pull request by klaren**
Configuration option `dbms.allow_format_migration` has been renamed to `dbms.allow_upgrade` since it applies to more than format migration. 

The expected behaviour is that the database should refuse to start if any changes will be made to the store that will prevent the database from starting on a previous version of neo4j.

### CODE REVIEW

File: community/neo4j/src/test/java/db/DatabaseStartupTest.java
Comment (burqen) [inline community/neo4j/src/test/java/db/DatabaseStartupTest.java:84]: Nice catch!

File: community/kernel/src/main/java/org/neo4j/kernel/configuration/GraphDatabaseConfigurationMigrator.java
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/kernel/configuration/GraphDatabaseConfigurationMigrator.java:104]: Nice! Can we also add a test for this?

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseSettings.java:133]: Have you looked into what implications this change has for docs? Let's make sure that is handled before merging.

**Review by burqen — CHANGES_REQUESTED** [review]

File: community/kernel/src/main/java/org/neo4j/kernel/configuration/GraphDatabaseConfigurationMigrator.java
Comment (klaren) [reply community/kernel/src/main/java/org/neo4j/kernel/configuration/GraphDatabaseConfigurationMigrator.java:104]: @burqen We definitely should, new commit available!

**Review by burqen — APPROVED** [review]

**Comment by burqen** [comment]
@klaren Approved! Don't know if the docs_compile error is related. It does not look like it, but worth checking with docs team before merging.

**Pull request merged**
